src/main/java/com/minecolonies/coremod/client/gui/WindowBuildBuilding.java

```diff
@@ -0,0 +1,362 @@
+            {
+                continue;
+            }
+
+            if (block != null
+                    && block != Blocks.AIR
+                    && !AbstractEntityAIStructure.isBlockFree(block, 0)
+                    && block != ModBlocks.blockSolidSubstitution
+                    && block != ModBlocks.blockSubstitution)
+            {
+                if (wrapper.getBlockInfo().tileentityData != null)
+                {
+                    final List<ItemStack> itemList = new ArrayList<>();
+                    if (wrapper.getBlockInfo() != null && wrapper.getBlockInfo().tileentityData != null)
+                    {
+                        itemList.addAll(ItemStackUtils.getItemStacksOfTileEntity(wrapper.getBlockInfo().tileentityData, world));
+                    }
+
+                    for (final ItemStack stack : itemList)
+                    {
+                       addNeededResource(stack, 1);
+                    }
+                }
+
+                addNeededResource(BlockUtils.getItemStackFromBlockState(blockState), 1);
+            }
+        }
+        window.findPaneOfTypeByID(LIST_RESOURCES, ScrollingList.class).refreshElementPanes();
+        updateResourceList();
+    }
+
+    /**
+     * Add a new resource to the needed list.
+     *
+     * @param res    the resource.
+     * @param amount the amount.
+     */
+    public void addNeededResource(@Nullable final ItemStack res, final int amount)
+    {
+        if (ItemStackUtils.isEmpty(res) || amount == 0)
+        {
+            return;
+        }
+        ItemStorage resource = resources.get(res.getUnlocalizedName());
+        if (resource == null)
+        {
+            resource = new ItemStorage(res);
+            resource.setAmount(amount);
+        }
+        else
+        {
+            resource.setAmount(resource.getAmount() + amount);
+        }
+        resources.put(res.getUnlocalizedName(), resource);
+    }
+
+    /**
+     * Initialise the previous/next and drop down list for style.
+     */
+    private void initStyleNavigation()
+    {
+        registerButton(BUTTON_PREVIOUS_STYLE_ID, this::previousStyle);
+        registerButton(BUTTON_NEXT_STYLE_ID, this::nextStyle);
+        stylesDropDownList = findPaneOfTypeByID(DROPDOWN_STYLE_ID, DropDownList.class);
+        stylesDropDownList.setHandler(this::onDropDownListChanged);
+        stylesDropDownList.setDataProvider(new DropDownList.DataProvider()
+        {
+            @Override
+            public int getElementCount()
+            {
+                return styles.size();
+            }
+
+            @Override
+            public String getLabel(final int index)
+            {
+                if (index >= 0 && index < styles.size())
+                {
+                    return styles.get(index);
+                }
+                return "";
+            }
+        });
+    }
+
+    /**
+     * called every time one of the dropdownlist changed.
+     *
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "list". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1172)